source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -136,11 +128,19 @@ InstanceImpl::ThreadLocalPool::threadLocalActiveClient(Upstream::HostConstShared
     client->host_ = host;
     client->redis_client_ = parent_.client_factory_.create(host, dispatcher_, parent_.config_);
     client->redis_client_->addConnectionCallbacks(*client);
+    // TODO(hyang): should the auth command and readonly command be moved to the factory method?
     if (!auth_password_.empty()) {
       // Send an AUTH command to the upstream server.
       client->redis_client_->makeRequest(Common::Redis::Utility::makeAuthCommand(auth_password_),
                                          null_pool_callbacks);
     }
+    // Any connection to replica requires the READONLY command in order to perform read.
+    // Also the READONLY command is a no-opt for the master.
+    // We only need to send the READONLY command iff it's possible that the host is a replica.
```

Comment:
I think this is a valid conclusion after looking at the redis 4 server code.

Reply:
Thanks for verifying